### DISCUSSION TITLE
达梦数据库同步数据库结构时，即便是数据库结构没有变化也创建了FTmp表

### DISCUSSION THREAD

**Discussion by EsonXie** [dotnetcore/FreeSql#1614] [Q&A]
数据库是达梦8
一个创建了FTmp表，如果发生异常，DDL语句的执行就停止了，然后重启的时候，重新创建FTmp表的时候，就报错了。。。
而且达梦数据库执行DDL语句时，即便是套了事务，也无法回滚。。。

我觉得是不是要想办法做一下数据库结构的对比，然后减少DDL语句的长度，让报错概率更小。。。

**Comment by 2881099** [comment]
表结构变化，sqlserver事务也不能回滚

**Comment by 2881099** [comment]
如果表起初使用freesql创建，后续对比成功率高很多。

另外codefirst一般只增，不改，不删，这样也不容易报错。ps 其他语言的项目开发也是如此。

**Reply by EsonXie** [reply]
我看现在的逻辑是创建临时表 然后再吧数据写入临时表
然后删除旧表，重命名临时表为新表
现在的问题是前一次同步结构的时候 发生异常，但临时表已经创建了，下一次同步时，会导致无法创建临时表。。。。

**Reply by EsonXie** [reply]
我再想是不是可以考虑临时表的表名改为每次同步都不一样的话，能不能解决这个问题
